packages/scaffold/src/partials/w3m-tooltip/index.ts

```diff
@@ -0,0 +1,68 @@
+import { html, LitElement } from 'lit'
+import { TooltipController } from '@web3modal/core'
+import { customElement } from '@web3modal/ui'
+import { state } from 'lit/decorators.js'
+import styles from './styles.js'
+
+@customElement('w3m-tooltip')
+export class W3mTooltip extends LitElement {
+  public static override styles = [styles]
+
+  // -- Members ------------------------------------------- //
+  private unsubscribe: (() => void)[] = []
+
+  // -- State & Properties -------------------------------- //
+  @state() private open = TooltipController.state.open
+
+  @state() private message = TooltipController.state.message
+
+  @state() private triggerRect = TooltipController.state.triggerRect
+
+  @state() private variant = TooltipController.state.variant
+
+  public constructor() {
+    super()
+    this.unsubscribe.push(
+      ...[
+        TooltipController.subscribe(newState => {
+          this.open = newState.open
+          this.message = newState.message
+          this.triggerRect = newState.triggerRect
+          this.variant = newState.variant
+        })
+      ]
+    )
+  }
+
+  // -- Lifecycle ----------------------------------------- //
+  public override disconnectedCallback() {
+    this.unsubscribe.forEach(unsubscribe => unsubscribe())
+  }
+
+  // -- Render -------------------------------------------- //
+  public override render() {
+    this.dataset['variant'] = this.variant
+
+    const topValue = this.triggerRect.top
+    const leftValue = this.triggerRect.left
+
+    this.style.cssText = `
+    --w3m-tooltip-top: ${topValue}px;
+    --w3m-tooltip-left: ${leftValue}px;
+    --w3m-tooltip-parent-width: ${this.triggerRect.width / 2}px;
+    --w3m-tooltip-display: ${this.open ? 'flex' : 'none'};
+    --w3m-tooltip-opacity: ${this.open ? 1 : 0};
```

Comment:
Bit of a nitpick:

I think we usually prefix these with `local`. 
I think w3m would be a bit confusing to know if it's a generic style vs a local one when modifying the css

Reply:
This is a good point. I've thought the same but the `w3m-tooltip` is a global component in w3m. Like snackbar and rendered on the root. So there should be only one tooltip rendered, that's why I wanted to use `--w3m-tooltip-*` naming.